airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -394,7 +392,8 @@ def execute(self, context):
                             retval = subprocess.check_output(args=[
                                 'kubectl', 'logs', pod_name, container_name])
 
-            self.clean_up(job_name)
             return retval
-        except Exception:
-            raise
+        finally:
+            self.log_container_logs(job_name, pod_output=pod_output)
+
+            self.clean_up(job_name)
```

Comment:
This will delete jobs that failed. This line should probably be outside the `finally`